core/controllers/collection_editor.py

```diff
@@ -197,8 +219,33 @@ def put(self, collection_id):
 
         # TODO(bhenning): Implement other rights changes here.
         is_public = self.payload.get('is_public')
-
-        if is_public is not None:
+        is_publicized = self.payload.get('is_publicized')
+        new_member_username = self.payload.get('new_member_username')
+        new_member_role = self.payload.get('new_member_role')
+        viewable_if_private = self.payload.get('viewable_if_private')
+
+        if new_member_username:
+            if not rights_manager.Actor(
+                    self.user_id).can_modify_roles(
+                        feconf.ACTIVITY_TYPE_COLLECTION,
+                        collection_id):
+                raise self.UnauthorizedUserException(
+                    'Only an owner of this collection can add or change '
+                    'roles.')
+
+            new_member_id = user_services.get_user_id_from_username(
+                new_member_username)
+            if new_member_id is None:
+                raise Exception(
+                    'Sorry, we could not find the specified user.')
+
+            rights_manager.assign_role_for_collection(
+                self.user_id, collection_id, new_member_id, new_member_role)
+            email_manager.send_role_notification_email(
```

Comment:
Please check -- does the email that gets sent actually refer to a "collection" rather than an exploration? I seem to remember the existing emails are for explorations. May want to liaise with @prasanna08 to see what modifications are required here.